### ISSUE TITLE
Backport Cluster goes to unavailable state after performing a `docker stop` on etcd node that is the first in `etcd-servers` api server argument

### ISSUE THREAD

**Issue by alena1108** [rancher/rancher#20454]
Backport https://github.com/rancher/rancher/issues/20311

**Comment by alena1108** [comment]
Can be tested with `v2.2.4-rc12`

**Comment by sangeethah** [comment]
Tested with `v2.2.4-rc12`.

Create a custom cluster.
Register node1 (all roles). Wait for cluster to become active.
Register node2 (all roles). Wait for cluster to become active.
Register node3 (all roles). Wait for cluster to become active.

Check that there are NO missing Ip and DNS entries in the etcd certs.
All 3 etcd nodes have  Ip and DNS entries for all the 3 nodes as expected.

```
root@ip-172-31-0-202:/etc/kubernetes/ssl# for f in kube-etc*.pem; do echo $f;  openssl x509 -text -noout -in $f | grep -i ip-172; done
kube-etcd-172-31-0-202-key.pem
unable to load certificate
140093405722264:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:701:Expecting: TRUSTED CERTIFICATE
kube-etcd-172-31-0-202.pem
                DNS:ip-172-31-0-202, DNS:ip-172-31-0-212, DNS:ip-172-31-15-118, DNS:localhost, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, IP Address:18.191.202.13, IP Address:172.31.0.202, IP Address:13.58.47.206, IP Address:172.31.0.212, IP Address:13.58.117.246, IP Address:172.31.15.118, IP Address:127.0.0.1, IP Address:10.43.0.1
kube-etcd-172-31-0-212-key.pem
unable to load certificate
140118419715736:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:701:Expecting: TRUSTED CERTIFICATE
kube-etcd-172-31-0-212.pem
                DNS:ip-172-31-0-202, DNS:ip-172-31-0-212, DNS:ip-172-31-15-118, DNS:localhost, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, IP Address:18.191.202.13, IP Address:172.31.0.202, IP Address:13.58.47.206, IP Address:172.31.0.212, IP Address:13.58.117.246, IP Address:172.31.15.118, IP Address:127.0.0.1, IP Address:10.43.0.1
kube-etcd-172-31-15-118-key.pem
unable to load certificate
140637376038552:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:701:Expecting: TRUSTED CERTIFICATE
kube-etcd-172-31-15-118.pem
                DNS:ip-172-31-0-202, DNS:ip-172-31-0-212, DNS:ip-172-31-15-118, DNS:localhost, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, IP Address:18.191.202.13, IP Address:172.31.0.202, IP Address:13.58.47.206, IP Address:172.31.0.212, IP Address:13.58.117.246, IP Address:172.31.15.118, IP Address:127.0.0.1, IP Address:10.43.0.1
root@ip-172-31-0-202:/etc/kubernetes/ssl# 
```

**Comment by soumyalj** [comment]
Tested with v2.2.4-rc12
Created DO cluster with 3 etcd nodes and one control/worker node
Deleted etcd nodes and a new node automatically gets created. After new node is created, verified that the command `openssl x509 -text -noout -in kube-etcd-a-b-c-d.pem` returns IP and DNS entries of all the 3 etcd nodes as expected. 
Cluster was operational and new workloads got created successfully.

Created AWS custom cluster with 3 etcd nodes and one control/worker node
Performed docker stop and docker start on the etcd nodes. 
Verified that the command `openssl x509 -text -noout -in kube-etcd-a-b-c-d.pem` returns IP and DNS entries of all the 3 etcd nodes as expected. 
Cluster was operational and new workloads got created successfully.

**Comment by sangeethah** [comment]
Reproduced the issue with RKE version -  v0.2.3-rc2

Create 1 node - node1 cluster (all roles).
Add 1 more node - node2 (./rke up) by adding the new node as the first node in the list in nodes in cluster.yml .
Add 1 more node - node3 (./rke up) by adding the new node as the first node in the list in nodes in cluster.yml .

Log in to node3.

Notice that there are Missing SANS for node's ip address entries.
Stop docker in the node.

```
root@ip-172-31-1-85:/etc/kubernetes/ssl# for f in kube-etc*.pem; do echo $f;  openssl x509 -text -noout -in $f | grep -i 18; done
kube-etcd-13-59-124-41-key.pem
unable to load certificate
140270730401432:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:701:Expecting: TRUSTED CERTIFICATE
kube-etcd-13-59-124-41.pem
        Serial Number: 7143579951101768798 (0x6323181e0205f45e)
            Not After : May 21 18:57:24 2029 GMT
                DNS:localhost, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, IP Address:13.59.124.41, IP Address:18.218.33.27, IP Address:18.219.150.127, IP Address:127.0.0.1, IP Address:10.43.0.1
         2e:f7:b8:f7:14:18:0a:d5:6f:ed:f7:0b:57:29:4e:b1:24:2e:
kube-etcd-18-218-33-27-key.pem
unable to load certificate
140248530011800:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:701:Expecting: TRUSTED CERTIFICATE
kube-etcd-18-218-33-27.pem
            Not After : May 21 18:54:20 2029 GMT
                DNS:localhost, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, IP Address:18.218.33.27, IP Address:18.219.150.127, IP Address:127.0.0.1, IP Address:10.43.0.1
kube-etcd-18-219-150-127-key.pem
unable to load certificate
140022791386776:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:701:Expecting: TRUSTED CERTIFICATE
kube-etcd-18-219-150-127.pem
                DNS:localhost, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, IP Address:18.219.150.127, IP Address:127.0.0.1, IP Address:10.43.0.1
         05:96:ff:8a:c2:3b:18:cf:86:d1:a1:9e:30:77:df:97:7c:1d:
root@ip-172-31-1-85:/etc/kubernetes/ssl# docker inspect 6e08ba5 | grep etcd-server
            "--etcd-servers=https://13.59.124.41:2379,https://18.218.33.27:2379,https://18.219.150.127:2379",
                "--etcd-servers=https://13.59.124.41:2379,https://18.218.33.27:2379,https://18.219.150.127:2379",
root@ip-172-31-1-85:/etc/kubernetes/ssl# sudo systemctl stop docker
root@ip-172-31-1-85:/etc/kubernetes/ssl# 
```

Cluster is broken. Any kubectl command timesout.
```
$ kubectl get nodes

Error from server (Timeout): the server was unable to return a response in the time allotted, but may still be processing the request (get nodes)

```

Issue is fixes with RKE version -  v0.2.3-rc4

Create 1 node - node1 cluster (all roles).
Add 1 more node - node2 (./rke up) by adding the new node as the first node in the list in nodes in cluster.yml .
Add 1 more node - node3 (./rke up) by adding the new node as the first node in the list in nodes in cluster.yml .

Log in to node3.

Notice that there are no Missing SANS for node's ip address entries.
```
root@ip-172-31-13-244:/etc/kubernetes/ssl# for f in kube-etc*.pem; do echo $f;  openssl x509 -text -noout -in $f | grep -i DNS; done
kube-etcd-18-188-55-116-key.pem
unable to load certificate
139923247232664:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:701:Expecting: TRUSTED CERTIFICATE
kube-etcd-18-188-55-116.pem
                DNS:localhost, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, IP Address:18.216.245.85, IP Address:18.188.55.116, IP Address:3.17.164.32, IP Address:127.0.0.1, IP Address:10.43.0.1
kube-etcd-18-216-245-85-key.pem
unable to load certificate
140094701352600:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:701:Expecting: TRUSTED CERTIFICATE
kube-etcd-18-216-245-85.pem
                DNS:localhost, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, IP Address:18.216.245.85, IP Address:18.188.55.116, IP Address:3.17.164.32, IP Address:127.0.0.1, IP Address:10.43.0.1
kube-etcd-3-17-164-32-key.pem
unable to load certificate
139942445991576:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:701:Expecting: TRUSTED CERTIFICATE
kube-etcd-3-17-164-32.pem
                DNS:localhost, DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, IP Address:18.216.245.85, IP Address:18.188.55.116, IP Address:3.17.164.32, IP Address:127.0.0.1, IP Address:10.43.0.1
root@ip-172-31-13-244:/etc/kubernetes/ssl# docker inspect kube-apiserver | grep etcd-servers
            "--etcd-servers=https://18.216.245.85:2379,https://3.17.164.32:2379,https://18.188.55.116:2379",
                "--etcd-servers=https://18.216.245.85:2379,https://3.17.164.32:2379,https://18.188.55.116:2379",
root@ip-172-31-13-244:/etc/kubernetes/ssl# sudo systemctl stop docker
```
Stop docker in the node.

Cluster continues to be functional.  kubectl commands continue to work.
